packages/flutter/lib/src/cupertino/segmented_control.dart

```diff
@@ -169,14 +223,61 @@ class _SegmentedControlState<T> extends State<SegmentedControl<T>> {
   }
 
   void _onTap(T currentKey) {
-    if (currentKey != widget.groupValue) {
+    if (currentKey != widget.groupValue && currentKey == _pressedKey) {
       widget.onValueChanged(currentKey);
+      _pressedKey = null;
+    }
+  }
+
+  Color getTextColor(int index, T currentKey) {
+    if (_selectionControllers.elementAt(index).isAnimating) {
```

Comment:
fwiw, it's more idiomatic to spell `elementAt` with square brackets, as in: `_selectionControllers[index].isAnimating`